### PR TITLE
Fix dtype mismatches for GPUs without bfloat16 support

### PR DESCRIPTION
**Pull request by kuff**
## What

Fix dtype mismatches that cause crashes when running on GPUs without bfloat16 support (compute capability < 8.0, e.g. RTX 20xx, GTX 16xx).

## Why

The pipeline assumes bfloat16 throughout. When models are converted to float16 for older GPUs, three categories of dtype mismatch cause `RuntimeError`:

1. **TimestepEmbedder** hardcodes `float()` for frequency embeddings, but MLP weights may be float16 → `mat1 and mat2 must have the same dtype`
2. **Noise tensors** are created as float32 via `torch.randn().to(device)` without matching the model's dtype → mismatches at the first linear layer
3. **Normalization statistics** (mean/std) default to float32 via `torch.tensor(list)` → arithmetic with float16 latent tensors fails

Additionally, `BiRefNet` (used for background removal) is incompatible with transformers 5.x due to meta device initialization calling `.item()`.

## Changes

| File | Change |
|------|--------|
| `trellis2/models/sparse_structure_flow.py` | Cast timestep frequency embeddings to MLP weight dtype before forward pass |
| `trellis2/pipelines/trellis2_image_to_3d.py` | Cast noise tensors to model dtype (5 sites) and normalization stats to latent dtype (5 sites) |
| `trellis2/pipelines/rembg/BiRefNet.py` | Runtime monkey-patch for transformers 5.x: add missing `all_tied_weights_keys` and skip meta device init |

All changes are backward-compatible — on GPUs with bfloat16 support, dtypes resolve to bfloat16 which matches existing behavior.

## Tested on

- **GPU**: NVIDIA RTX 2060 (6GB VRAM, SM 7.5, no bfloat16)
- **Model**: `microsoft/TRELLIS.2-4B` (smallest variant)
- **Parameters**: `pipeline_type='512'`, `max_num_tokens=32768`, render resolution 512
- **Precision**: `convert_module_to_f16` for flow models (preserves float32 norms), `image_cond_model` kept in float32 (overflows in fp16)
- Pipeline runs end-to-end: sparse structure → shape → texture → decode → render → GLB export

### Sample output
https://github.com/user-attachments/assets/95b0b6ab-8655-42be-a218-df33e7580f59

Generated from `assets/example_image/T.png` on RTX 2060 6GB:

> **Note**: To reproduce, the caller must convert models to float16 via `convert_module_to_f16` and override `model.dtype` from bfloat16 to float16. This PR only fixes the library-side dtype assumptions — the caller-side setup is demonstrated in a [separate example script](https://github.com/kuff/TRELLIS.2/blob/feat/6gb-vram-example/example_6gb.py).

### CODE REVIEW

**Comment by kuff** [comment]
> @kuff the command you issued was incorrect. Please try again.
> 
> Examples are:
> 
> ```
> @microsoft-github-policy-service agree
> ```
> 
> and
> 
> ```
> @microsoft-github-policy-service agree company="your company"
> ```

@microsoft-github-policy-service agree company="Aalborg University"

**Comment by kuff** [comment]
For those interested, the code for 6gb inference is available at https://github.com/kuff/TRELLIS.2/blob/feat/6gb-vram-example/example_6gb.py

**Comment by kuff** [comment]
Closed because I realize the noise changes could break inference on the original target GPUs and are thus unmergable and in fact not backwards compatible as originally stated.
